Source/automap.cpp

```diff
@@ -775,7 +776,10 @@ std::unique_ptr<AutomapTile[]> LoadAutomapData(size_t &tileCount)
 {
 	switch (leveltype) {
 	case DTYPE_TOWN:
-		return LoadFileInMem<AutomapTile>("levels\\towndata\\automap.amp", &tileCount);
+		if (*sgOptions.Graphics.showTownMap)
+			return LoadFileInMem<AutomapTile>("levels\\towndata\\automap.amp", &tileCount);
+		else
+			return nullptr;

```

Comment:
I don't think there is a need for this as it won't have an effect when the map isn't being rendered any way.
```suggestion
		return LoadFileInMem<AutomapTile>("levels\\towndata\\automap.amp", &tileCount);
```